site/static/api/device-registry-v1.yaml

```diff
@@ -0,0 +1,724 @@
+               items:
+                  $ref: '#/components/schemas/Adapter'
+            "limits":
+               $ref: '#/components/schemas/ResourceLimit'
+            "trusted-ca":
+               $ref: '#/components/schemas/TrustedCA'
+
+      TrustedCA:
+         type: object
+         additionalProperties: false
+         required:
+            - "subject-dn"
+         properties:
+            "subject-dn":
+               type: string
+            "public-key":
+               type: string
+               format: byte
+            "cert":
+               type: string
+               format: byte
+
+      Adapter:
+         type: object
+         additionalProperties: true
+         required:
+            - type
+         properties:
+            "type":
+               type: string
+            "enabled":
+               type: boolean
+               default: false
+            "device-authentication-required":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+
+      ResourceLimit:
+         type: object
+         additionalProperties: false
+         properties:
+            "max-connections":
+               type: integer
+               default: -1
+
+# Devices schema
+
+      Device:
+         type: object
+         additionalProperties: false
+         properties:
+            "enabled":
+               type: boolean
+               default: true
+            "ext":
+               $ref: '#/components/schemas/Extensions'
+
+# Credentials
+
+      CredentialsSet:
+         type: array
+         items:
+            $ref: '#/components/schemas/TypedSecret'
+
+      TypedSecret:
+         oneOf:
+            - $ref: '#/components/schemas/PasswordSecret'
+            - $ref: '#/components/schemas/PSKSecret'
+            - $ref: '#/components/schemas/X509CertificateSecret'
+         discriminator:
+            propertyName: type
+            mapping:
+               "hashed-password": '#/components/schemas/PasswordSecret'
+               "psk": '#/components/schemas/PSKSecret'
+               "x509-cert": '#/components/schemas/X509CertificateSecret'
+
+      CommonSecret:
+         type: object
+         required:
+            - auth-id
+            - type
+         properties:
+            "type":
+               type: string
+            "auth-id":
+               type: string
+            "enabled":
+               type: boolean
+               default: true
+            "not-before":
+               type: string
+               format: date-time
+            "not-after":
+               type: string
+               format: date-time
+            "comment":
+               type: string
+
+      X509CertificateSecret:
+         allOf:
+            - $ref: '#/components/schemas/CommonSecret'
+
+      PasswordSecret:
```

Comment:
can this secret have additional properties?

Reply:
Not to my knowledge.

Reply:
See above.